SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -463,17 +463,18 @@
           AND $(Language) == 'C#' 
           ">true</SonarQubeRunRoslynCodeAnalysis>
       <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeRunRoslynCodeAnalysis) == '' ">false</SonarQubeRunRoslynCodeAnalysis>
-    </PropertyGroup>
 
+    </PropertyGroup>
+    
     <PropertyGroup Condition=" $(SonarQubeRunRoslynCodeAnalysis) == 'true' ">
       <ErrorLog Condition=" $(ErrorLog) == '' " >$(TargetDir)SonarQube.Roslyn.ErrorLog.json</ErrorLog>
       <ResolvedCodeAnalysisRuleSet>$(SonarQubeRoslynRulesetFullName)</ResolvedCodeAnalysisRuleSet>
     </PropertyGroup>
 
     <ItemGroup Condition=" $(SonarQubeRunRoslynCodeAnalysis) == 'true' ">
-      <!-- Remove any existing references to SonarLint -->
-      <Analyzer Remove="@(Analyzer)" Condition=" $([System.String]::new('%(Analyzer.Identity)').EndsWith('SonarLint.dll', System.StringComparison.OrdinalIgnoreCase)) == 'true'" />
-      <Analyzer Include="$(SonarLintAnalyzerAssembly)"/>
+      <!-- Remove any existing references to files called *SonarLint*.dll -->
+      <Analyzer Remove="@(Analyzer)" Condition="$([System.Text.RegularExpressions.Regex]::IsMatch($([System.String]::new('%(Analyzer.Identity)')), 'SonarLint[^\\]*\.dll$', System.Text.RegularExpressions.RegexOptions.IgnoreCase)) " />
+      <Analyzer Include="$(SonarLintAnalyzerDir)*SonarLint*.dll" />
```

Comment:
I wanted to avoid hard-coding a list of SonarLint assemblies if possible. I'm assuming there will always be a file called SonarLint.dll* and I'm using that as a marker to check for the existence of the analyzer assemblies.
I'm then using looser name matching to remove existing and add new assemblies: any assembly in SonarLint in the name will be matched.